oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java

```diff
@@ -384,11 +384,21 @@ public APIGatewayProxyResponseEvent handleRequest(
                 boolean setAuthenticatedFlagForIPV =
                         configurationService.isAuthenticatedFlagForIpvEnabled();
 
+                CredentialTrustLevel lowestRequestedCredentialTrustLevel =
+                        VectorOfTrust.getLowestCredentialTrustLevel(clientSession.getVtrList());
+                long authTime = NowHelper.now().toInstant().getEpochSecond();
                 if (setAuthenticatedFlagForIPV) {
-                    sessionService.storeOrUpdateSession(
-                            userSession.setNewAccount(accountState).setAuthenticated(true));
-                    orchSessionService.updateSession(
-                            orchSession.withAuthTime(NowHelper.now().toInstant().getEpochSecond()));
+                    if (userSession.isAuthenticated()) {
+                        if (lowestRequestedCredentialTrustLevel.compareTo(
+                                        userSession.getCurrentCredentialStrength())
+                                > 0) {
+                            orchSessionService.updateSession(orchSession.withAuthTime(authTime));
+                        }
+                    } else {
+                        sessionService.storeOrUpdateSession(
+                                userSession.setNewAccount(accountState).setAuthenticated(true));
+                        orchSessionService.updateSession(orchSession.withAuthTime(authTime));
+                    }
```

Comment:
Do you think there is much value in writing this a bit inverted like this:

```
 long authTime = NowHelper.now().toInstant().getEpochSecond();
                if (setAuthenticatedFlagForIPV) {

                    if(!userSession.isAuthenticated()){
                        orchSessionService.updateSession(orchSession.withAuthTime(authTime));
                    } else if (userSession.isAuthenticated() &&  isUpliftJourney){
                        orchSessionService.updateSession(orchSession.withAuthTime(authTime));
                    }
                } else {
                    sessionService.storeOrUpdateSession(userSession.setNewAccount(accountState));
                }
```

Reply:
Have updated in the latest commit